### PR TITLE
Added LO-TO form of PoleResidue model

### PR DESCRIPTION
**Pull request by caseyflex**
Simple functions to generate PoleResidue model from LO-TO form.

### CODE REVIEW

File: tidy3d/components/medium.py [hunk 1]
```diff
@@ -2045,6 +2045,92 @@ def to_medium(self) -> Medium:
             frequency_range=self.frequency_range,
         )
 
+    @classmethod
+    def lo_to_eps_model(
+        cls,
```
Comment (tylerflex) [inline tidy3d/components/medium.py:2050]: does this need to be a `classmethod` or would a `staticmethod` be sufficient?

**Review by tylerflex — COMMENTED** [review]
looks fine to me, just a couple comments.

File: tidy3d/components/medium.py [hunk 2]
```diff
@@ -2045,6 +2045,92 @@ def to_medium(self) -> Medium:
...
+        """
+
+        omegas_lo, gammas_lo, omegas_to, gammas_to = (np.array(param) for param in zip(*poles))
+
+        # discriminants of quadratic factors of denominator
+        discs = 2 * np.emath.sqrt((gammas_to / 2) ** 2 - omegas_to**2)
+
+        # require nondegenerate TO poles
+        if len({(omega_to, gamma_to) for (_, _, omega_to, gamma_to) in poles}) != len(poles) or any(
+            disc == 0 for disc in discs
```
Comment (weiliangjin2021) [inline tidy3d/components/medium.py:2094]: When will the 1st condition occur?

**Review by weiliangjin2021 — APPROVED** [review]

**Comment by tomflexcompute** [comment]
Thanks @caseyflex for the implementation. I tested it on MoO3 with the literature result and it worked out greatly. Not super important but since the LOTO form of the permittivity is usually considered as a specific form of the Lorentz model, does it make sense to have it as td.Lorentz.from_loto instead of from the PoleResidue?

File: tidy3d/components/medium.py (hunk 2 above)
Comment (caseyflex) [reply tidy3d/components/medium.py:2094]: The first condition will occur when a TO pole appears repeatedly within the provided list of poles. Then, the set will be smaller than the list of poles provided.

File: tidy3d/components/medium.py [hunk 3]
```diff
@@ -2045,6 +2045,92 @@ def to_medium(self) -> Medium:
...
+            The LO-TO poles, given as list of tuples of the form
+            (omega_LO, gamma_LO, omega_TO, gamma_TO).
+
+        Returns
+        -------
+        :class:`.PoleResidue`
+            The pole residue equivalent of the LO-TO form provided.
+        """
+
+        omegas_lo, gammas_lo, omegas_to, gammas_to = (np.array(param) for param in zip(*poles))
```
[the remark this answers is not in the document]
Comment (caseyflex) [reply tidy3d/components/medium.py:2087]: thanks!

[the remark this answers is not in the document]
Comment (caseyflex) [reply tidy3d/components/medium.py:2087]: actually turned out to be ```omegas_lo, gammas_lo, omegas_to, gammas_to = map(np.array, zip(*poles))```

[the remark this answers is not in the document]
Comment (tylerflex) [reply tidy3d/components/medium.py]: actually, if the params are all the same size, you might be able to simply do
```py
omegas_lo, gammas_lo, omegas_to, gammas_to = np.array(poles)
```

**Review by tomflexcompute — APPROVED** [review]

File: tidy3d/components/medium.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (caseyflex) [reply tidy3d/components/medium.py]: that didn't seem to work...

[the remark this answers is not in the document]
Comment (tylerflex) [reply tidy3d/components/medium.py]: no worries then

**Pull request merged**